src/NServiceBus.Core/Routing/IRouteSource.cs

```diff
@@ -0,0 +1,11 @@
+namespace NServiceBus
+{
+    using System;
+    using Routing;
+
+    interface IRouteSource
+    {
+        void GenerateRoutes(Action<RouteTableEntry> registerRouteCallback);
```

Comment:
can't we make this `IEnumerable<RouteTableEntry> GenerateRoutes(Conventions conventions)`?